### PR TITLE
Check Path in Graph

### PR DESCRIPTION
**Pull request by ishwari20**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given an undirected graph G(V, E) and two vertices v1 and v2 (as integers), check if there exists any path between them or not. Print true if the path exists and false otherwise.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #5272

### CODE REVIEW

File: C-Plus-Plus/graphs/checkPath.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    
+    bool small=path(V, mat, visited, v1, v2);
+    
+    if(small){
+        cout<<"true";
+    }
+    else{
+        cout<<"false";
+    }
+
```
Comment (bhav09) [inline C-Plus-Plus/graphs/checkPath.cpp]: Please remove redundant lines.

File: C-Plus-Plus/graphs/checkPath.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+4 4
+0 1
+0 3
+1 2
+2 3
+1 3
+Sample Output 1 :
+true
+
+Time Complexity: O(V+E)
```
Comment (bhav09) [inline C-Plus-Plus/graphs/checkPath.cpp]: Time complexity should be above Sample I/O

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed. 
Need to add more comments to the code.
Try to make main function as a "driver function" only.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Can you comment on the logic along with the code? 
Make sure, the comments are above the code lines.
Rest perfect.

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -253,8 +253,12 @@
 - [Shortest Path in Unweighted Graph](graphs/shortest_path_using_bfs.cpp)
 - [Depth first Search for a graph](graphs/dfs_of_a_graph.cpp)
 - [Diameter of Binary Tree](graphs/Diameter_of_binary_tree.cpp)
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove this line

File: C-Plus-Plus/README.md [hunk 4]
```diff
@@ -253,8 +253,12 @@
...
+- [Check Path in Graph](graphs/checkPath.cpp)
+
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove this line

File: C-Plus-Plus/graphs/checkPath.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        cout<<"false";
+    }
+    return 0;
+}
+
+
+/*
+Time Complexity: O(V+E)
+
+Space Complexity: O(V^2)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/checkPath.cpp]: Tell what `V` and `E` here signify

File: C-Plus-Plus/graphs/checkPath.cpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        return true;
+    }  
+    visited[v1]=true;
+    
+    for(int i=0; i<V; i++){
+        if(visited[i]){
+            continue;
+        }
+        // visited[i]=true;
+	//If this adjacent node is the destination node, then return true
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/checkPath.cpp]: Indent the comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format the entire code

**Comment by bhav09** [comment]
@ishwari20, There are a few unresolved issues in this PR. Please resolve them.

File: C-Plus-Plus/graphs/checkPath.cpp [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+Sample Input 1 :
+4 4
+0 1
+0 3
+1 2
+2 3
+1 3
+Sample Output 1 :
+true
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/graphs/checkPath.cpp]: Use proper display message how you are reading and writing outputs

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes mentioned.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**